array/array_test.go

```diff
@@ -88,11 +89,11 @@ func TestString(t *testing.T) {
 			if want, got := len(tc.want)+2, b.Cap(); want != got {
 				t.Errorf("unexpected builder cap -want/+got:\n\t- %d\n\t+ %d", want, got)
 			}
-			mem.AssertSize(t, tc.sz)
+			assert.Equal(t, tc.sz, mem.CurrentAlloc(), "unexpected memory allocation.")
```

Comment:
@jsternberg FYI, this is the change in Arrow's checked allocator that I was talking about. `AssertSize` now contains a check that ensures that there are no outstanding allocations, and also that the numeric argument matches the cumulative amount of memory allocated. It doesn't seem like it should do that based on the name, so this was a surprising change.

Reply:
I think this indicates an error. I'll download the PR and take a look. It does not seem like `AssertSize` was changed based on the code.

I'm basing it off of a quick reading of these sections:

https://github.com/apache/arrow/blob/56e270fda7f5/go/arrow/memory/checked_allocator.go#L131-L134
https://github.com/apache/arrow/blob/56e270fda7f5/go/arrow/memory/checked_allocator.go#L42
https://github.com/apache/arrow/blob/56e270fda7f5/go/arrow/memory/checked_allocator.go#L73

Reply:
Ah, so `AssertSize` still checks the current allocations (not total) but it also seems to only work when the argument is zero since if the argument wasn't zero there should be "leaked" allocations. But, they're not leaked.